test/jdk/javax/swing/JFrame/MaximizeWindowTest.java

```diff
@@ -0,0 +1,53 @@
+import javax.swing.JFrame;
+import javax.swing.JSplitPane;
+import javax.swing.SwingUtilities;
+import java.awt.Dimension;
+import java.awt.Robot;
+import java.awt.event.ComponentAdapter;
+import java.awt.event.ComponentEvent;
+import java.lang.reflect.InvocationTargetException;
+
+/*
+ * @test
+ * @key headful
+ * @summary setExtendedFrame not executed immediately
+ * @run main MaximizeWindowTest
+ */
+@SuppressWarnings("serial")
+public class MaximizeWindowTest extends JFrame
+{
+    public static void main(String[] arguments) throws Exception {
+        Robot robot = new Robot();
+
+        SwingUtilities.invokeAndWait(new Runnable() {
```

Comment:
Suggestion, can you use the following instead of overriding the run method.
` SwingUtilities.invokeAndWait(() -> {
            
        });`